dns_sd_avahi.c

```diff
@@ -92,42 +125,37 @@ static void __avahi_resolver_cb(AvahiServiceResolver *resolver,
 							resolver))));
 		break;
 	case AVAHI_RESOLVER_FOUND: {
-		/* Avahi is multi-threaded, so lock the list */
-		iio_mutex_lock(ddata->lock);
-		ddata->resolved++;
-
-		/* Find empty data to store things*/
-		while (ddata->next)
-			ddata = ddata->next;
-
-		/* link a new placeholder to the list */
-		avahi_address_snprint(ddata->addr_str,
-				sizeof(ddata->addr_str), address);
-		memcpy(ddata->address, address, sizeof(*address));
-		ddata->port = port;
-		ddata->hostname = strdup(host_name);
-		ddata->resolved = true;
-		/* link a new, empty placeholder to the list */
-		if (!new_discovery_data(&ddata->next)) {
-			/* duplicate poll & lock info,
-			 * since we don't know which might be discarded */
-			ddata->next->poll = ddata->poll;
-			ddata->next->lock = ddata->lock;
-		} else {
-			IIO_ERROR("Avahi Resolver : memory failure\n");
-		}
-		iio_mutex_unlock(ddata->lock);
-
```

Comment:
this move looks like it's own commit

Reply:
Agreed.

Reply:
Well, I still think this is simple enough that it could go in the same patch but it's already 2 complaining about this so I have to resign myself and do it in it's own patch :sweat_smile:

Reply:
It's a good habit to create many small commits with atomic changes rather than a big one. Especially if you have people (maintainers) review your code. Here in particular, it's not obvious at first sight that this code didn't change and was just moved, and it makes the actual functionality change appear much bigger than it should be.